### DISCUSSION TITLE
🚨 Dependabot alerts REST API is now available in public beta

### DISCUSSION THREAD

**Discussion by erinhav** [community/community#34231] [Code Security]
You can now programmatically view and act on Dependabot alerts via the REST API. New endpoints to [view, list, and update Dependabot alerts](https://docs.github.com/en/rest/dependabot/alerts) are available in a public beta. This release covers alerts in a repository-scoped list. We're working on some follow-up ships, including organization and enterprise-level lists.

How are you using the Dependabot alerts API? What feedback do you have for us? Please let us know -- we ❤️ your feedback!

**Comment by tjcorr** [comment]
One thing that would be nice is if the REST API include the id of each alert to provide consistency with graphql (the id under the node object) and the node_id of the webhook event `repository_vulnerability_alert`. Right now the only shared identifier is the number field.

**Comment by SalimBensiali** [comment]
Is being able to update other fields on dependabot alerts in scope in the future? Currently one is only able to dismiss or reopen a dependabot alert.
I would love to be able t programmatically trigger a dependabot pull request creation similar to what is being done on the Github UI.
Also I would love to be to be able to set in additional metadata on a dependabot alert.

**Reply by erinhav** [reply]
👋 what kind of fields/metadata would you want to update?

**Reply by SalimBensiali** [reply]
To begin with, being able to update the scope would suffice in my current use case.

**Reply by t4nguy3n** [reply]
@erinhav  Could you help review this question? https://github.com/orgs/community/discussions/61417 Thank you

**Comment by rainbow1016** [comment]
very good question!

**Comment by Nimarrakers** [comment]
> You can now programmatically view and act on Dependabot alerts via the REST API. New endpoints to [view, list, and update Dependabot alerts](https://docs.github.com/en/rest/dependabot/alerts) are available in a public beta. This release covers alerts in a repository-scoped list. We're working on some follow-up ships, including organization and enterprise-level lists.
> 
> How are you using the Dependabot alerts API? What feedback do you have for us? Please let us know -- we ❤️ your feedback!

**Comment by wildan3105** [comment]
Hi @erinhav, is there any roadmap to add API to get the statistics of dependabot alerts on top of the current APIs? Essentially, I want to get a formatted data via API that can be displayed like this page: `https://github.com/orgs/:org_name/security/risk` (how many open and closed security issues, etc.)

Appreciate any response to this 🙇

**Reply by kellyarwine** [reply]
Hey @wildan3105,

I appreciate your message! We're on the brink of launching a security overview dashboard that'll shed more light on your Dependabot alerts, alongside code scanning and secret scanning alerts. After gathering feedback and refining, we're planning to release an API to back the dashboard. This will allow you to fetch summary counts without downloading each alert individually.

Look out for the dashboard release at Universe, and anticipate the API roll-out in the first quarter of 2024. 

If you'd like to see a preview of the dashboard, I'd love to show it to you.   [Book a session](deleted) for an exclusive walkthrough of the security overview dashboard and as a thank you, I'll ensure you're among the first to access it during its private beta launch. _NOTE: I will circle back and delete this last paragraph in 7 days to avoid spam._

**Comment by Miguel-Sanz** [comment]
Exciting updates! We've introduced a new feature, and we're eager to hear your thoughts.

**Comment by coopernetes** [comment]
Hi all, just a quick note. `GET /enterprises/{enterprise}/dependabot/alerts` returns an empty array. I have the correct permissions (org owner on multiple orgs in an Enterprise instance) and still cannot view alerts according to the docs. Thanks!

https://docs.github.com/en/enterprise-cloud@latest/rest/dependabot/alerts?apiVersion=2022-11-28#list-dependabot-alerts-for-an-enterprise

**Comment by dustincox-rouse** [comment]
Hi @erinhav, We could really use the re-opened the re-opened date on the GHAS side. For example on the UI we have "dependabot reopened this ... last week. Otherwise it's throwing off our vulnerability management since the original time it saw it was years ago. Is there any way we can add in the last reopened date? Thank you

**Reply by carogalvin** [reply]
@dustincox-rouse thanks for that feedback! We're not actively working in this area right now, but I've logged your request for the future!

**Reply by lux-capacitor** [reply]
Would also like this made available, seeing a dependabot alert saying "500 days old" because it resurfaced a week ago is pretty frustrating. If this is expected to be of any use to enterprise I need to know when an item came back up, not when it first ever appeared anywhere.

**Comment by Lewa4444** [comment]
@tjcorr

**Comment by mattsalves** [comment]
The release of the **Dependabot** alerts REST API marks a significant step towards enhancing the security and stability of our development ecosystems. This programmatic integration not only streamlines dependency **management** but also fosters a culture of automation and **accountability**, empowering teams to proactively respond to vulnerabilities and keep their projects up-to-date and resilient.

**Comment by WaterSwallow** [comment]
> You can now programmatically view and act on Dependabot alerts via the REST API. New endpoints to [view, list, and update Dependabot alerts](https://docs.github.com/en/rest/dependabot/alerts) are available in a public beta. This release covers alerts in a repository-scoped list. We're working on some follow-up ships, including organization and enterprise-level lists.
> 
> How are you using the Dependabot alerts API? What feedback do you have for us? Please let us know -- we ❤️ your feedback!

The news about viewing and acting on Dependabot alerts via the REST API is fantastic. This new feature could streamline the process of handling alerts for many developers and organizations. With the ability to work with these alerts programmatically, developers can more effectively manage their projects' dependencies.

The introduction of new endpoints to view, list, and update Dependabot alerts is a significant step forward, particularly with their coverage for repository-scoped lists. It's also great to hear about the upcoming plans for expanding this to organization and enterprise-level lists.

As always, collecting user feedback is crucial for continual improvement. Encouraging users to share their experiences and feedback with the Dependabot alerts API is crucial. I'm certain that many users will find these enhancements beneficial and look forward to seeing further improvements based on user feedback. 

If anyone has specific use cases or feedback regarding this feature, don't hesitate to share. Your feedback is always appreciated!
